libraries/AP_Param/AP_Param.h

```diff
@@ -542,11 +542,16 @@ class AP_ParamT : public AP_Param
             set(v);
         }
     }
-    
+
     /// Value setter - set value, tell GCS
     ///
     void set_and_notify(const T &v) {
+// We do want to compare each value, even floats, since it being the same here
+// is the result of previously setting it.
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wfloat-equal"
         if (v != _value) {
+#pragma GCC diagnostic pop
```

Comment:
Ok, so we only have two options that I can see:

- ignore the warning like you are doing here
- change it to use float epsilon like below

@tridge?